documentation/modules/deploying/proc-deploy-migrate-kraft.adoc

```diff
@@ -0,0 +1,256 @@
+* The Cluster Operator that manages the Kafka cluster is running.
+* The Kafka cluster deployment uses Kafka node pools.
++
+If your ZooKeeper-based cluster is already using node pools, it is ready to migrate.
+If not, you can xref:proc-migrating-clusters-node-pools-str[migrate the cluster to use node pools]. 
+To migrate, brokers must be contained in a `KafkaNodePool` resource configuration that is assigned a `broker` role and has the name `kafka`.
+Support for node pools is enabled in the `Kafka` resource configuration using the `strimzi.io/node-pools: enabled` annotation.
+
+In this procedure, the Kafka cluster name is `my-cluster`, which is located in `my-project`. 
+The name of the controller node pool is `controller`.
+
+.Procedure
+
+. For the Kafka cluster, create a node pool with a `controller` role.
++
+Add a quorum of controller nodes to the node pool.
++
+.Example configuration for a controller node pool
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaNodePoolApiVersion}
+kind: KafkaNodePool
+metadata:
+  name: controller
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  roles:
+    - controller
+  storage:
+    type: jbod
+    volumes:
+      - id: 0
+        type: persistent-claim
+        size: 20Gi
+        deleteClaim: false
+    resources:
+      requests:
+        memory: 64Gi
+        cpu: "8"
+      limits:
+        memory: 64Gi
+        cpu: "12"    
+----
++
+NOTE: For the migration, you cannot use a node pool of nodes that share the broker and controller roles.
+
+. Apply the new `KafkaNodePool` resource to create the controllers.
++
+Errors related to using controllers in a ZooKeeper-based environment are expected.
+
+. Enable KRaft migration in the `Kafka` resource by setting the `strimzi.io/kraft` annotation to `migration`:
++
+[source,shell]
+----
+kubectl annotate kafka my-cluster strimzi.io/kraft: migration
+----
++
+.Enabling KRaft migration
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+  namespace: my-project
+  annotations:
+    strimzi.io/kraft: migration 
+# ...
+----
++
+Applying the annotation to the `Kafka` resource configuration starts the migration.
+
+. Check the controllers have started and the brokers have rolled:
++
+[source,shell]
+----
+kubectl get pods -n my-project
+----
++
+.Output shows nodes in broker and controller node pools
+[source,shell]
+----
+NAME                     READY  STATUS   RESTARTS
+my-cluster-kafka-0       1/1    Running  0
+my-cluster-kafka-1       1/1    Running  0
+my-cluster-kafka-2       1/1    Running  0
+my-cluster-controller-3  1/1    Running  0
+my-cluster-controller-4  1/1    Running  0
+my-cluster-controller-5  1/1    Running  0
+# ...
+----
++
+Here, the brokers have the name `my-cluster-kafka` as they were contained in a node pool named `kafka` when migrating the cluster to use node pools.
```

Comment:
"when migrating the cluster to use node pools" ... well "if migrating ... " 
See my comment above, it's not always true. People could have cluster using node pools for brokers since the beginning and they didn't have the need to migrate (so the pool had a name different from `kafka`).

Reply:
It's not always true and I think we're clear on that early on. But, as Jakub pointed out, we're using the migration from no node pools as that would likely be the starting point. I don't think we need to repeat what we said earlier here.

Reply:
Tbh I am not sure on which basis we can say users will start from no node pools. Do we have any evidence of that. Can't we have users having ZooKeeper with node pools already? I think so. My opinion is having the documentation not biased on that assumption but the other way around and adding that "if you are not using node pools already, please migrate to use them first".